src/context/product-context.jsx

```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext, useReducer } from "react";
+import { ProductReducer } from "../reducers/product-reducer";
+
+const ProductContext = createContext();
+
+const ProductProvider = ({ children }) => {
+	const [productState, dispatch] = useReducer(ProductReducer, {
+		productsList: [],
+		sortBy: "",
+		rating: "",
```

Comment:
here you can use null, not sure but I think its good practice saw in many peoples code.

Reply:
alright tushar ! will change